thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineTaskRunner.java

```diff
@@ -138,8 +137,12 @@ public DetectionPipelineTaskRunner() {
         this.evaluationDAO.save(evaluationDTO);
       }
 
-      // run maintenance flow to update model
-      config = maintenanceFlow.maintain(config, Instant.now());
+      try {
+        // run maintenance flow to update model
+        config = maintenanceFlow.maintain(config, Instant.now());
+      } catch (Exception e) {
+        LOG.warn("Re-tune pipeline {} failed", config.getId(), e);
+      }
```

Comment:
If the maintain failed then what is the config?
We should generate a new config and replace the old one if maintain() succeeds.

Reply:
if it maintain() failed, the config will be kept the same. The tuning only updates the "componentSpecs" of the detection config. If succeeds, it will create a new "componentSpecs" and replace the old one.